core/sdk-core/src/main/java/software/amazon/awssdk/core/sync/RequestBody.java

```diff
@@ -220,15 +221,21 @@ public static RequestBody fromContentProvider(ContentStreamProvider provider, lo
     }
 
     /**
-     * Creates a {@link RequestBody} from the given {@link ContentStreamProvider}.
+     * Creates a {@link RequestBody} from the given {@link ContentStreamProvider} when the content length is unknown. If you
+     * are able to provide the content length at creation time, consider using {@link #fromInputStream(InputStream, long)} or
+     * {@link #fromContentProvider(ContentStreamProvider, long, String)} to negate the need to read through the stream to find
```

Comment:
Can we also link to https://docs.aws.amazon.com/AmazonS3/latest/API/s3_example_s3_Scenario_UploadStream_section.html as a recommendation to upload stream with unknown content length?